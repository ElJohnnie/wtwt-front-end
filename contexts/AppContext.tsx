import React, {
  createContext,
  useState,
  useRef,
  useCallback,
  ReactNode,
  FC,
} from 'react';
import { AppContextProps, Step, Questions } from '../types';

const defaultState = {
  steps: { current: [] as Step[] },
  currentStep: 0,
  questions: [],
  answeredSteps: [] as boolean[],
  answers: [] as string[],
  nextStep: () => {
    throw new Error('nextStep function not implemented');
  },
  prevStep: () => {
    throw new Error('prevStep function not implemented');
  },
  answerQuestion: () => {
    throw new Error('answerQuestion function not implemented');
  },
  showResult: false,
};

export const AppContext = createContext<AppContextProps>(defaultState);

interface AppProviderProps {
  children: ReactNode;
}

export const AppProvider: FC<AppProviderProps> = ({ children }) => {
  const steps = useRef<Step[]>([
    { number: '1', hasRange: true, isInactive: false, asComplete: false },
    { number: '2', hasRange: true, isInactive: true, asComplete: false },
    { number: '3', hasRange: true, isInactive: true, asComplete: false },
    { number: '4', hasRange: true, isInactive: true, asComplete: false },
    { number: '5', hasRange: false, isInactive: true, asComplete: false },
  ]);

  const questions: Questions[] = [
    {
      question: 'Como você está se sentindo agora?',
      answers: [
        { text: 'Feliz' },
        { text: 'Triste' },
        { text: 'Ansioso' },
        { text: 'Bem animado' },
        { text: 'Entediado' },
      ],
      description: 'teste 123',
    },
    {
      question: 'Qual o seu estilo de filme favorito?',
      answers: [
        { text: 'Ação' },
        { text: 'Aventura' },
        { text: 'Animação' },
        { text: 'Comédia' },
        { text: 'Drama' },
        { text: 'Fantasia' },
        { text: 'SciFi' },
        { text: 'Thriller' },
        { text: 'Terror/Suspense' },
      ],
      description: 'teste 123',
    },
    {
      question: 'Seu segundo estilo de filme favorito?',
      answers: [
        { text: 'Ação' },
        { text: 'Aventura' },
        { text: 'Animação' },
        { text: 'Comédia' },
        { text: 'Drama' },
        { text: 'Fantasia' },
        { text: 'SciFi' },
        { text: 'Thriller' },
        { text: 'Terror/Suspense' },
      ],
      description: 'teste 123',
    },
    {
      question: 'Qual periodo de lançamento mais lhe convém?',
      answers: [
        { text: '2000' },
        { text: '1990' },
        { text: '1980' },
        { text: '1970' },
        { text: '1960' },
        { text: '1950' },
        { text: '1940' },
      ],
      description: 'teste 123',
    },
  ];

  const [currentStep, setCurrentStep] = useState<number>(0);

  const [showResult, setShowResult] = useState<boolean>(false);

  const [answeredSteps, setAnsweredSteps] = useState<boolean[]>([
    false,
    false,
    false,
    false,
    false,
  ]);
  const [answers, setAnswers] = useState<string[]>([]);

  const nextStep = useCallback(() => {
    setCurrentStep((prev) => Math.min(prev + 1, steps.current.length - 1));
  }, []);

  const prevStep = useCallback(() => {
    setCurrentStep((prev) => Math.max(prev - 1, 0));
  }, []);

  const answerQuestion = useCallback(
    (text: string) => {
      if (currentStep < questions.length - 1) {
        setAnswers((prev) => {
          const newAnswers = [...prev];
          newAnswers[currentStep] = text;
          return newAnswers;
        });
        setAnsweredSteps((prev) => {
          const newAnsweredSteps = [...prev];
          newAnsweredSteps[currentStep] = true;
          return newAnsweredSteps;
        });
        nextStep();
      } else {
        setAnswers((prev) => {
          const newAnswers = [...prev];
          newAnswers[currentStep] = text;
          return newAnswers;
        });
        setAnsweredSteps((prev) => {
          const newAnsweredSteps = [...prev];
          newAnsweredSteps[currentStep] = true;
          return newAnsweredSteps;
        });
        setShowResult(true);
      }
    },
    [currentStep, nextStep, questions.length],
  );

  return (
    <AppContext.Provider
      value={{
        steps,
        currentStep,
        questions,
        answeredSteps,
        answers,
        nextStep,
        prevStep,
        answerQuestion,
        showResult,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};
