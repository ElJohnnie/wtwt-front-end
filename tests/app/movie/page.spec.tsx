import React from 'react';
import '@testing-library/jest-dom';
import HomePage from '../../../app/movie/page';
import { AppContext } from '../../../contexts/AppContext';
import { renderWithOutContextProvider } from '../../../utils/tests/helpers';

jest.mock('lottie-react', () => ({
  __esModule: true,
  default: () => <div>Lottie Mock</div>,
}));

const redirectMock = jest.fn();
const replaceMock = jest.fn();

jest.mock('../../../hooks/useNavigation/index.tsx', () => ({
  ...jest.requireActual('../../../hooks/useNavigation/index.tsx'),
  useNavigation: () => ({
    redirect: redirectMock,
    replace: replaceMock,
  }),
}));

const mockContextValue = {
  steps: {
    current: [
      { number: '1', hasRange: true, isInactive: false, asComplete: false },
      { number: '2', hasRange: true, isInactive: true, asComplete: false },
      { number: '3', hasRange: true, isInactive: true, asComplete: false },
      { number: '4', hasRange: true, isInactive: true, asComplete: false },
      { number: '5', hasRange: false, isInactive: true, asComplete: false },
    ],
  },
  currentStep: 0,
  questions: [
    {
      question: 'Como você está se sentindo agora?',
      answers: [
        { text: 'Feliz', value: 'happy' },
        { text: 'Triste', value: 'sad' },
        { text: 'Ansioso', value: 'anxious' },
        { text: 'Bem animado', value: 'excited' },
        { text: 'Entediado', value: 'bored' },
      ],
      description: 'teste 123',
    },
  ],
  answeredSteps: [false, false, false, false, false],
  answers: [],
  nextStep: jest.fn(),
  prevStep: jest.fn(),
  answerQuestion: jest.fn(),
  showResult: false,
  resetState: jest.fn(),
};

describe('Home', () => {
  it('Renderizando a página inicial', () => {
    const { container } = renderWithOutContextProvider(
      <AppContext.Provider value={mockContextValue}>
        <HomePage />
      </AppContext.Provider>,
    );

    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
