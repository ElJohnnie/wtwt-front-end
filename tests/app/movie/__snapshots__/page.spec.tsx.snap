// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renderizando a página inicial 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center"
  >
    <div
      class="flex justify-between flex-row items-center"
    >
      <div
        class="MuiBox-root css-1tylbmg"
      >
        <div
          class="MuiStepper-root MuiStepper-horizontal css-1a9jkgz-MuiStepper-root"
        >
          <div
            class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-nwqhv8-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    1
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-nwqhv8-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-nwqhv8-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-nwqhv8-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    4
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-nwqhv8-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    5
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center my-12"
    >
      <div
        class="py-2 px-2 md:px-12 rounded-full text-center"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-12vu8f8-MuiTypography-root"
        >
          Como você está se sentindo agora?
        </h2>
      </div>
    </div>
    <div
      class="my-4 flex flex-wrap gap-4 justify-center max-w-2xl"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Feliz
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Triste
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Ansioso
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Bem animado
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Entediado
      </button>
    </div>
    <div
      class="flex items-center my-12"
    >
      <div
        class="text-center text-slate-700"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          teste 123
        </p>
      </div>
    </div>
    <div
      class="flex flex-row items-center justify-center w-6/12 space-x-4"
    >
      <div
        class="hidden"
      >
        <div
          class="w-full flex flex-col justify-center items-center"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1r976ha-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
              data-testid="UndoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
